containers/cell-type-annotations-to-factors/context/main.py

```diff
@@ -0,0 +1,64 @@
+import argparse
+from glob import glob
+from pathlib import Path
+from os import mkdir, environ
+import json
+
+import pandas as pd
+from enum import Enum
+
+
+class COLUMNS(Enum):
+    # Cell Type Annotation column names
+    CELL_ID = "cell_id"
+    ANNOTATION = "annotation"
+    PREDICTION_SCORE = "prediction_score"
+
+
+def cell_type_annotations_to_factors(input_path, output_path):
+    df = pd.read_csv(input_path)
+    
+    # Get a list containing all unique annotation values in the input file.
+    annotation_values = df[COLUMNS.ANNOTATION.value].unique().tolist()
+
+    # Append a temporary column to the dataframe containing the index of each annotation,
+    # relative to the `annotation_values` list.
+    df["annotation_index"] = df[COLUMNS.ANNOTATION.value].apply(lambda val: annotation_values.index(val))
+    
+    # Construct the factors dict, which will be saved as a JSON object.
+    factors = {
+        "Cell Type Annotations": {
+            "map": annotation_values,
+            "cells": dict(zip(df[COLUMNS.CELL_ID.value].tolist(), df["annotation_index"].tolist()))
+        }
+    }
+    pretty_json_factors = json.dumps(factors).replace('}},', '}},\n')
```

Comment:
I don't think this actually does anything - I think you can use options for json dumps to get a output print if you want it.

Reply:
True, thanks, just updated